### DISCUSSION TITLE
AOSP 12.1 v413

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#2298] [General]
- May security patch level
- Fix fingerprint enrolling on some Samsung devices (Notably A53 and S22)
- [Samsung settings] Control flashlight intensity
- Fix OTA for some devices (see https://github.com/phhusson/treble_experimentations/issues/2289 )
- Switch fingerprint-on-display to white, since some devices require it

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v413'>AOSP 12.1 v413</a>.</em>

**Comment by yaseen5050** [comment]
I'm trying to connect to a network that has mac authentication and it won't connect. I want to use the phone mac not randomized mac but there is no setting in the wifi menu to use the phone mac

**Reply by Hausemaster8281** [reply]
> I'm trying to connect to a network that has mac authentication and it won't connect. I want to use the phone mac not randomized mac but there is no setting in the wifi menu to use the phone mac

Options to turn it off are there in developer options

**Reply by Hausemaster8281** [reply]
> I'm trying to connect to a network that has mac authentication and it won't connect. I want to use the phone mac not randomized mac but there is no setting in the wifi menu to use the phone mac

Specifically under Networking ![Screenshot_20220504-185613_Settings~2.png](https://user-images.githubusercontent.com/66825701/166690838-52dad2d4-9127-4bd0-8df5-b7e42b4d883d.png)

**Reply by yaseen5050** [reply]
I tried disabling that but it didn't work

**Comment by danzj7** [comment]
Does v413 fixes recent button on arm32 goapps variant?

**Reply by devrieshh** [reply]
No,only on the vanilla version it works

**Comment by navinkhatasana** [comment]
Sir Please remove **search bar** and **adjusted padding** in _original aosp 12.1 launcher_

**Comment by Jafery20** [comment]
Hello! Good day, with this latest version I have had problems with the fingerprint on the screen, due to the white color that is now used... Is there a way to change it to green, as in the previous version? Thank you very much for your attention!

**Comment by iotY** [comment]
@phhusson , security patch: `sed -n '/PLATFORM_.*:= [0-9]/p' build/core/version_defaults.mk` is still on 2022-03-05

**Reply by phhusson** [reply]
v413's manifest.xml has `<project name="platform/build" path="build/make" revision="e344c83b927e286d8871e5bd3d373da326fbaa4e" upstream="refs/tags/android-12.1.0_r5" dest-branch="refs/tags/android-12.1.0_r5" groups="pdk">`
https://android.googlesource.com/platform/build/+/refs/tags/android-12.1.0_r5/core/version_defaults.mk#243
say   `PLATFORM_SECURITY_PATCH := 2022-05-05`

**Reply by iotY** [reply]
Sorry I "slept" my `repo init` ...

Thx again for ...everything.

**Comment by pranav-vaid** [comment]
@phhusson 
Is it a good idea to flash a new updated vbmeta(if available) before flashing this?
I am currently on v412. 
Are there any extra steps for updating vbmeta or do we only need to flash vbmeta file and reboot?

**Reply by JonnyTech** [reply]
No, phh recommends stock vendor for his GSI.

**Reply by pranav-vaid** [reply]
Sorry, i was confusing vendor with vbmeta
Fixed it.

**Reply by pranav-vaid** [reply]
@JonnyTech what do you mean by stock vendor here?
Phh's GSI vendor, or the vendor that came with device ROM?

**Reply by JonnyTech** [reply]
A GSI is a Generic SYSTEM Image - it is flashed on the phone's system partition only, you still need your phone's stock vendor untouched. A GSI is not a ROM. GSI = system. ROM = system. vendor, product, etc. Therefore, flash your phone's full stock ROM them flash the GSI to the system partition.

**Reply by pranav-vaid** [reply]
I understand the concept of GSI @JonnyTech 
I just want to know how to proceed if i have already installed a GSI.

Scenario: After installing a GSI, my phone's manufacturer might have provided vendor(or any other updates, except system). What should i do in that case because re-flashing new update of my stock ROM, then again flashing GSI is not an option, right?

**Reply by JonnyTech** [reply]
When my phone manufacturer releases a new update, I download it from their site and flash from fastboot everything except system and userdata.

**Reply by pranav-vaid** [reply]
Alright. Will try that.

**Comment by Tura05** [comment]
Hi guys
 1.) I wanted to know which version to choose for my Samsung A6
![Screenshot_20220513-073712_Device Info](https://user-images.githubusercontent.com/74953162/168234681-ba750cc3-8497-4514-8960-2cf1ada83370.jpg)

 2.) Is it possible to flash it via TWRP?

**Comment by devrieshh** [comment]
The 32bit versions

[system-squeak-arm32_binder64-ab-gogapps.img.xz]
 [system-squeak-arm32_binder64-ab-vanilla.img.xz]
 [system-squeak-arm32_binder64-ab-vndklite-gogapps-secure.img.xz]
 [system-squeak-arm32_binder64-ab-vndklite-gogapps.img.xz]
 [system-squeak-arm32_binder64-ab-vndklite-vanilla-secure.img.xz]
 [system-squeak-arm32_binder64-ab-vndklite-vanilla.img.xz]

normaly you install it in fastboot(i dont know with samsung phones)    you can install it with twrp When you have that for your phone

**Comment by Cosmiika** [comment]
seems like changing the fod color to white breaks it in a few other devices aswell, maybe best to make it toggle-able in phh treble settings?

**Comment by pranav-vaid** [comment]
@phhusson when will this update be available on OTA channel?

**Reply by phhusson** [reply]
I won't push this version as OTA, because it has a FOD regression (because it is now white instead of green, which not all devices support), and FLOSS variant has its webview and web browser broken.

**Reply by pranav-vaid** [reply]
Ahh, no problem. Thanks.

**Comment by d3zd3** [comment]
I just found out that I have a problem with trying to OTA do v413. And it spits out a error that update couldn't be downloaded. But the download size is missing.![Screenshot_20220516-152357_PhhTrebleApp.png](https://user-images.githubusercontent.com/77751911/168603002-32523b02-4072-486e-81fd-9d768d4f15de.png)![Screenshot_20220516-152400_PhhTrebleApp.png](https://user-images.githubusercontent.com/77751911/168603025-26c44e7e-1930-4b16-8ffc-f3ba74733073.png)

**Reply by phhusson** [reply]
I... didn't put v413 to OTA? Is this my unmodified AOSP GSI?

**Reply by d3zd3** [reply]
> I... didn't put v413 to OTA? Is this my unmodified AOSP GSI?

Ah yep my bad sorry.

**Reply by d3zd3** [reply]
> I... didn't put v413 to OTA? Is this my unmodified AOSP GSI?

and by the way will you add option to the OTA app to install updates from local package stored on a device?

**Comment by albertordzcz** [comment]
@phhusson is the volte and no sound on calling fixed on this build? my device a moto edge x30 SD 8 gen 1. really thanks for your hard work!

**Comment by Grdnskii** [comment]
i had problems with connecting to internet using mobile data.
is there any cure for this problem?

**Comment by devrieshh** [comment]
Is it possible to let the 'overview button'  work in the GSI other than the vanille ones where they work already?i am using a 32bit [Cubot King Kong Mini 2]
